jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PlatformChecker.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.builder.steps;
+
+import com.google.cloud.tools.jib.api.LogEvent;
+import com.google.cloud.tools.jib.api.buildplan.Platform;
+import com.google.cloud.tools.jib.configuration.BuildContext;
+import com.google.cloud.tools.jib.event.EventHandlers;
+import com.google.cloud.tools.jib.image.json.ContainerConfigurationTemplate;
+import com.google.common.base.Verify;
+import java.util.Set;
+
+/** Provides helper methods to check platforms. */
+public class PlatformChecker {
+
+  /**
+   * Assuming the base image is not a manifest list, checks and warns misconfigured platforms.
+   *
+   * @param buildContext the {@link BuildContext}
+   * @param containerConfig container configuration JSON of the base image
+   */
+  static void checkManifestPlatform(
+      BuildContext buildContext, ContainerConfigurationTemplate containerConfig) {
+    EventHandlers eventHandlers = buildContext.getEventHandlers();
+    String baseImageName =
+        buildContext.getBaseImageConfiguration().getTarPath().isPresent()
+            ? buildContext.getBaseImageConfiguration().getTarPath().get().toString()
+            : buildContext.getBaseImageConfiguration().getImage().toString();
+
+    Set<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
+    Verify.verify(!platforms.isEmpty());
+
+    if (platforms.size() != 1) {
+      eventHandlers.dispatch(
+          LogEvent.warn(
+              "platforms configured, but '" + baseImageName + "' is not a manifest list"));
+    } else {
+      Platform platform = platforms.iterator().next();
+      if (!platform.getArchitecture().equals(containerConfig.getArchitecture())
+          || !platform.getOs().equals(containerConfig.getOs())) {
+
+        // Unfortunately, "platforms" has amd64/linux by default even if the user didn't explicitly
+        // configure it. Skip reporting to suppress false alarm.
+        if (!(platform.getArchitecture().equals("amd64") && platform.getOs().equals("linux"))) {
```

Comment:
Just making sure I'm understanding this correctly, so this check needs to be included because we don't have a way to differentiate between implicitly or explicitly configured platforms?

Reply:
Yes, when it's a single platform of `amd64/linux`. That is, even if the user doesn't configure anything, `platforms` is a singleton set with `amd64/linux`. If you think the code comment on line 56-57 can be improved, feel free to suggest an edit.